### ISSUE TITLE
Install Not Working on Centos 7 Server

### ISSUE THREAD

**Issue by nootkan** [pH7Software/pH7-Social-Dating-CMS#809]
I had this script working on my local wamp server and decided I liked what I saw and tried to do a clean install on my Linux Centos 7 server but keep getting a 404 error.  

I've read through the 47 issues, docs and wiki but didn't see anything that would help or maybe I missed something but read them a couple of times.

I've un-commented the rewrite base and added the folder inside two slashes (RewriteBase /dating/) and uploaded to server with filezilla.  I also made the 777 permission changes like suggested even though I have suphp installed so everything runs on 755.  Tried with permission changes and without and still no luck.

I ensured that mod_rewrite is enabled on my server with sudo httpd -M |grep rewrite_module and received the rewrite_module (shared) response to confirm.

Server Details:
CENTOS 7.8 [vanisle1] v88.0.13 
Apache Version | 2.4.46
PHP Version | 7.2.33
MySQL Version | 5.7.31
Architecture | x86_64
Perl Version | 5.16.3
Kernel Version | 3.10.0-1127.10.1.el7.x86_64

Server logs:
[Sun Aug 16 14:55:12.034868 2020] [:error] [pid 19343:tid 47760912946944] [client ipaddress:63070] SoftException in Application.cpp:267: File "/home/mywebsite/public_html/dating/_install/index.php" is writeable by group
[Sun Aug 16 14:55:12.034952 2020] [core:error] [pid 19343:tid 47760912946944] [client ipaddress:63070] End of script output before headers: index.php
[Sun Aug 16 14:58:36.575017 2020] [:error] [pid 19343:tid 47760815150848] [client ipaddress:63086] SoftException in Application.cpp:267: File "/home/mywebsite/public_html/dating/_install/index.php" is writeable by group
[Sun Aug 16 14:58:36.575103 2020] [core:error] [pid 19343:tid 47760815150848] [client ipaddress:63086] End of script output before headers: index.php
[Sun Aug 16 14:58:55.603818 2020] [:error] [pid 19343:tid 47760819353344] [client ipaddress:63100] SoftException in Application.cpp:267: File "/home/mywebsite/public_html/dating/_install/index.php" is writeable by group
[Sun Aug 16 14:58:55.603863 2020] [core:error] [pid 19343:tid 47760819353344] [client ipaddress:63100] End of script output before headers: index.php
[Sun Aug 16 15:15:03.349496 2020] [:error] [pid 19343:tid 47760802543360] [client ipaddress:63395] SoftException in Application.cpp:267: File "/home/mywebsite/public_html/dating/_install/index.php" is writeable by group
[Sun Aug 16 15:15:03.349582 2020] [core:error] [pid 19343:tid 47760802543360] [client 70.67.143.193:63395] End of script output before headers: index.php
[Sun Aug 16 15:16:33.466172 2020] [:error] [pid 19343:tid 47760838264576] [client ipaddress:63411] SoftException in Application.cpp:267: File "/home/mywebsite/public_html/dating/_install/index.php" is writeable by group
[Sun Aug 16 15:16:33.466271 2020] [core:error] [pid 19343:tid 47760838264576] [client 70.67.143.193:63411] End of script output before headers: index.php
[Sun Aug 16 15:19:20.150705 2020] [core:alert] [pid 19429:tid 47760838264576] [client ipaddress:63439] /home/mywebsite/public_html/dating/.htaccess: Invalid command 'RewriteBase /dating/', perhaps misspelled or defined by a module not included in the server configuration

**Comment by polynamaude** [comment]
@nootkan You have your error on the last line of your logs.
` Invalid command 'RewriteBase /dating/', perhaps misspelled or defined by a module not included in the server configuration`

I suggest you look at the Apache configuration docs for your version. This can be found at https://httpd.apache.org/

**Comment by polynamaude** [comment]
@nootkan 
https://httpd.apache.org/docs/2.4/mod/mod_rewrite.html#rewritebase

You have missing double quotes.

**Comment by nootkan** [comment]
Thanks but that didn't work either.  I have installed the script successfully on the same server under another domain name in the root instead of a folder so it has to be an htaccess issue.  Here is what I have in my htaccess file:
<IfModule mod_rewrite.c>
    <IfModule mod_negotiation.c>
        Options -MultiViews -Indexes
    </IfModule>
    Options +FollowSymLinks

    RewriteEngine On

    ## Uncomment the below "#RewriteBase /" (remove "#") if pH7CMS is installed in a folder, and add the folder name after the slash "/"
    ## Example: If pH7CMS is installed in "/public_html/ph7cms/", then it should be "RewriteBase /ph7cms/", however it may depends of your Apache configuration.
    ## If you need, please ask your hosting company or check the Apache doc: http://httpd.apache.org/docs/current/mod/mod_rewrite.html

    RewriteBase "/dating/"
</IfModule>

**Comment by polynamaude** [comment]
@nootkan So you've found where the problem reside...
Like it is said in the comment you may have to use the whole path reference or only the relative, depending on your Apache configuration.
I'd suggest you either work this out with your hosting company or look on the CentOS/Apache configuration forum as this doesn't have nothing to do with this software.

**Comment by nootkan** [comment]
Okay I got the install to start but received an error that says "allow_url_fopen PHP directive needs to be enabled" which begs the question as to why this is needed to run your script (security issue).  

When I installed the script in the root of another domain on the same server I never received this error which has me wondering if it is indeed required to run the install or dating script. 

I did enable allow_url_fopen on the cpanel account but the error still remains.

Thanks for your replies and support.

**Comment by nootkan** [comment]
I've narrowed this down to a permissions issue.  It seems that no matter what sftp client I use the permissions are being set to 775/664 instead of 755/644. After manually setting the permissions to the right ones I am able to start the installation process.  I now get the error that says I need to enable "allow_url_fopen" which I am currently trying to do for the subfolder only.  

Not asking for help just documenting my steps in case someone else has a similar issue.

**Comment by nootkan** [comment]
Still trying to figure out how to install inside a subfolder successfully.  Going to try to delete everything and start over.  Set "allow_url_fopen" to on but still getting the error.  Will update after re-uploading and trying a fresh install.

**Comment by Undefined-Variables** [comment]
All you have to do is create a folder in your root directory, go to domain.com/that-folder/_install/ and follow the install. 

You can now access that install at domain.com/that-folder

if you want to access it without using the folders name in the url you just add 

  ````  
    RewriteEngine On
    RewriteBase /that-folder/

````

to your .htaccess file found in that folder. Its that simple.

**Comment by Undefined-Variables** [comment]
If you are not able to access it in

    domain.com/that-folder 

try  

domain.com/that-folder/index.php

**Comment by nootkan** [comment]
Okay I finally got it installed and working correctly.  

The main problem I surmise that I had originally, was the recent cPanel update had changed my umask permissions for the domain on my server.  

After resetting the umask to default to 755/644 again I deleted all my folders and files and did a fresh install.  I then had to enable the allow_url_fopen globally as it wouldn't work using the MultiPHP INI Editor in the domains cPanel account.  

After the install was complete and everything was working normally, I disabled the allow_url_fopen again as it is security issue.  I then checked the script again and everything is working fine in both the website and admin areas.  

None of this would have been so difficult had my permissions been set to what they were supposed to be by my server.  The only steps I would have had to take was to enable the allow_url_fopen for the install and change the htaccess file to "rewritebase /subfolder/ as previously mentioned.  

Hopefully my experience will help someone else who may find themselves in a similar situation.  

Thanks to all who took the time out of their busy schedules to provide support and direction. Much appreciated.
